src/meta/src/hummock/compaction_group/manager.rs

```diff
@@ -108,15 +108,13 @@ impl<S: MetaStore> CompactionGroupManager<S> {
 
     /// Unregisters `table_fragments` from compaction groups
     pub async fn unregister_table_fragments(&self, table_fragments: &TableFragments) -> Result<()> {
-        let table_ids = table_fragments
-            .internal_table_ids()
-            .into_iter()
-            .chain(std::iter::once(table_fragments.table_id().table_id))
-            .collect_vec();
         self.inner
             .write()
             .await
-            .unregister(&table_ids, self.env.meta_store())
+            .unregister(
+                &table_fragments.all_table_ids().collect_vec(),
```

Comment:
How about letting `all_table_ids` return `Vec<>`?

Reply:
actually lots of situations it will not be collected to a vec

Reply:
But there will be a `IntoIterator`?

Reply:
I think `iter().flat_map()` is better than `iter().flat_map().collect().into_iter()`....

Reply:
I mean that `IntoIterator` will be called by rustc automatically

Reply:
automatically generate `iter().flat_map().collect().into_iter()` ... I am not sure if the rustc can optimize it.

Reply:
For example, now it returns `impl Iterator<Item = u32> + '_`. We will call it in these situations:
```rust
HashSet::from_iter(table_fragment.all_table_ids()) // Need IntoIterator

for table_id in table_fragment.all_table_ids() // Need IntoIterator

.flat_map(|table_fragments| table_fragments.all_table_ids()) // Need IntoIterator

&table_fragments.all_table_ids().collect_vec(), // Need Vec<>
```
When we change return type into `Vec<>`, the first three don't need to change. Rust will adapt `IntoIterator` requirement.

Reply:
I mean adaptation in code text, not optimization at instruction level.

Reply:
I don't think it's a big deal, either way is fine.